freemocap/core_processes/post_process_skeleton_data/calculate_center_of_mass.py

```diff
@@ -0,0 +1,383 @@
+    0.436,
+    0.430,
+    0.430,
+    0.506,
+    0.506,
+    0.433,
+    0.433,
+    0.433,
+    0.433,
+    0.5,
+    0.5,
+]
+
+segment_COM_percentages = [
+    0.081,
+    0.497,
+    0.028,
+    0.028,
+    0.016,
+    0.016,
+    0.006,
+    0.006,
+    0.1,
+    0.1,
+    0.0465,
+    0.0465,
+    0.0145,
+    0.0145,
+]
+
+
+# %%
+def calculate_segment_COM(
+        segment_conn_len_perc_dataframe,
+        skelcoordinates_frame_segment_joint_XYZ,
+        num_frame_range,
+):
+    segment_COM_frame_dict = []
+    for frame in track(num_frame_range, description="Calculating Segment Center of Mass"):
+        segment_COM_dict = {}
+        for segment, segment_info in segment_conn_len_perc_dataframe.iterrows():
+            this_segment_XYZ = skelcoordinates_frame_segment_joint_XYZ[frame][segment]
+
+            # for mediapipe
+            this_segment_proximal = this_segment_XYZ[0]
+            this_segment_distal = this_segment_XYZ[1]
+            this_segment_COM_length = segment_info["Segment_COM_Length"]
+
+            this_segment_COM = this_segment_proximal + this_segment_COM_length * (
+                    this_segment_distal - this_segment_proximal
+            )
+            segment_COM_dict[segment] = this_segment_COM
+        segment_COM_frame_dict.append(segment_COM_dict)
+    return segment_COM_frame_dict
+
+
+# %%
+
+
+def reformat_segment_COM(segment_COM_frame_dict, num_frame_range, num_segments):
+    segment_COM_frame_imgPoint_XYZ = np.empty([int(len(num_frame_range)), int(num_segments), 3])
+    for frame in num_frame_range:
+        this_frame_skeleton = segment_COM_frame_dict[frame]
+        for joint_count, segment in enumerate(this_frame_skeleton.keys()):
+            segment_COM_frame_imgPoint_XYZ[frame, joint_count, :] = this_frame_skeleton[segment]
+    return segment_COM_frame_imgPoint_XYZ
+
+
+# %%
+
+
+def calculate_total_body_COM(segment_conn_len_perc_dataframe, segment_COM_frame_dict, num_frame_range):
+    logger.info("Calculating Total Body Center of Mass")
+    totalBodyCOM_frame_XYZ = np.empty([int(len(num_frame_range)), 3])
+
+    for frame in track(num_frame_range, description="Calculating Total Body Center of Mass..."):
+
+        this_frame_total_body_percentages = []
+        this_frame_skeleton = segment_COM_frame_dict[frame]
+
+        for segment, segment_info in segment_conn_len_perc_dataframe.iterrows():
+            this_segment_COM = this_frame_skeleton[segment]
+            this_segment_COM_percentage = segment_info["Segment_COM_Percentage"]
+
+            this_segment_total_body_percentage = this_segment_COM * this_segment_COM_percentage
+            this_frame_total_body_percentages.append(this_segment_total_body_percentage)
+
+        this_frame_total_body_COM = np.nansum(this_frame_total_body_percentages, axis=0)
+
+        totalBodyCOM_frame_XYZ[frame, :] = this_frame_total_body_COM
+
+    f = 2
+    return totalBodyCOM_frame_XYZ
+
+
+# %%
+
+
+def build_anthropometric_dataframe(
```

Comment:
Can we save this dataframe out somewhere with a `to_csv()` or `to_json()` call? 


(see longer comment above for context)

...After looking at this again, I realize this function is more or less doing what I was requesting, that is it's generating a nice schema based on the anthropometic data defined as variables above and then using that to calculate the COM later (or something like that) 

I think what I'm proposing now is to just save this data out upon calculation - at first as a nice record of the schema used in the calculation, and later we might just have a folder of `json`s or `toml`s or something that includes various definitions of COM anthropometic definitions (e.g. from other papers, or adjust for prosthetics, etc) as well as schemas to define interfaces between a given anthropometric skeleton schema and a given skeleton data schema (e.g. mediapipe, qualisys, etc) 


Again - these are on the fly thoughts, I might be missing something